gradle/scripts/lib/java-publish.gradle

```diff
@@ -23,7 +23,16 @@ configure(projectsWithFlags('publish', 'java')) {
                     jarOverrideFile = tasks.trimShadedJar.outJarFiles.find() as File
                     jarOverrideTask = tasks.trimShadedJar
                 } else if (tasks.findByName('shadedJar')) {
-                    jarOverrideFile = tasks.shadedJar.archivePath
+                    def archivePath = tasks.shadedJar.archivePath
+                    if (project.hasFlags('scala')) {
+                        def microVersionIndex = project.ext.defaultScalaVersions.lastIndexOf('.')
+                        def scalaVersionSuffix = project.ext.defaultScalaVersions.substring(0, microVersionIndex)
```

Comment:
- Could we do a little bit more strict regular expression matching to extract the suffix, so that a poorly formatted version number is not accepted?
- Shouldn't we detect the Scala version from the `scala-library` dependency in `dependencies.yml`?